django/db/migrations/autodetector.py

```diff
@@ -17,6 +18,22 @@
     resolve_relation,
 )
 
+OperationDependency = namedtuple(
+    "OperationDependency", "app_label model_name field_name type"
+)
```

Comment:
I think we can simplify things further below by doing something like the following:

```python
class OperationDependency(
    namedtuple("OperationDependency", "app_label model_name field_name type")
):
    @cached_property
    def model_name_lower(self):
        return self.model_name.lower()

    @cached_property
    def field_name_lower(self):
        return self.field_name.lower()
```

This would allow changes such as the following below:

```diff
-                and operation.model_name_lower == dependency.model_name.lower()
-                and operation.name_lower == dependency.field_name.lower()
+                and operation.model_name_lower == dependency.model_name_lower
+                and operation.name_lower == dependency.field_name_lower
```

I'm not sure how often these comparisons occur, but it might save something?

It may also be worth nesting `DependencyType` as it's strictly related to `OperationDependency`:

```python
class OperationDependency(...):
    class Type(Enum):
        ...
    ...
```

And then use `OperationDependency.Type.CREATE`, etc?

Reply:
Updated.